src/org/mozilla/javascript/NativeArray.java

```diff
@@ -847,13 +848,17 @@ private void setLength(Object val) {
      * or its value is not convertible to a number.
      */
     static long getLengthProperty(Context cx, Scriptable obj) {
-        // These will both give numeric lengths within Uint32 range.
+        // These will give numeric lengths within Uint32 range.
         if (obj instanceof NativeString) {
             return ((NativeString)obj).getLength();
         }
         if (obj instanceof NativeArray) {
             return ((NativeArray)obj).getLength();
         }
+        if (obj instanceof XMLObject) {
+            Callable lengthFunc = (Callable) ((XMLObject) obj).get("length", obj);
+            return ((Number) lengthFunc.call(cx, obj, obj, ScriptRuntime.emptyArgs)).longValue();
```

Comment:
Is it possible that the result will be not a number, or out of range? There's code before to validate "doubleLen". Does it make sense to set "doubleLen" and use the rest of the logic in this function instead?

Reply:
XMLObjectImpl.length() returns an int.